### PR TITLE
Add changelog for v4.2.0

### PR DESCRIPTION
**Pull request by leiyiz**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add changelog for v4.2.0 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-4.2.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Changes by Kind
+
+### Feature
+
+- Logging for gRPC call can now be limited configurably with flag `--max-grpc-log-length` ([#411](https://github.com/kubernetes-csi/external-attacher/pull/411), [@leiyiz](https://github.com/leiyiz))
+
+## Dependencies
+
+### Added
+- cloud.google.com/go/compute: v1.12.1
```
Comment (leiyiz) [inline CHANGELOG/CHANGELOG-4.2.md:16]: although it's auto-generated, I don't see this in go.mod so not sure where this is coming from...

Comment (jsafrane) [reply CHANGELOG/CHANGELOG-4.2.md:16]: It got pulled by `google.golang.org/grpc@v1.52.3`. I did not dig deeper why grpc depends on google cloud libs.

File: CHANGELOG/CHANGELOG-4.2.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Release notes for v4.1.0
```
Comment (jsafrane) [inline CHANGELOG/CHANGELOG-4.2.md:1]: v4.2.0

Comment (leiyiz) [reply CHANGELOG/CHANGELOG-4.2.md]: done

**Comment by leiyiz** [comment]
@jsafrane I don't have permission to cut a new release, would you mind also cutting `v4.2.0` after this PR is merged? Much thanks!

**Pull request merged**